### PR TITLE
feat: edit/delete/reply for emoji/sticker/audio

### LINKED ISSUE #13246: Add missing actions (edit/delete/reply) to certain types of messages (emoji/sticker/audio)
# Feature Issue

Currently, certain types of chat messages (emoji/sticker/audio) are missing some actions (edit/delete/reply) that can be applicable to those types of messages but not yet implemented. Therefore, we should add those missing actions (see description below). 

### User Story

As a user, I want to have possibility to **edit/delete** _**emoji**_ messages that are sent by me in 1-1/group chats. 
As a user, I want to have possibility to **reply** to _**audiomessages**_ that are received/sent by me in 1-1/group chats. 
As a user, I want to have possibility to **reply** to **_sticker_** messages that are received/sent by me and **delete** this type of messages sent by me in 1-1/group/public chats. 

### Description

In 1-1/group chats:
1. **_emoji_**: add **edit/delete** actions (for messages sent by user)
2. _**audiomessage**_: add **reply** action
3. **_sticker_**: add **reply** action. For messages sent by user - add **delete** action.

In public chat:
1. **_sticker_**: add **reply** action. For messages sent by user - add **delete** action.

### PR DESCRIPTION
**Pull request by OmarBasem**
#### This PR is a continuation of that [PR](https://github.com/status-im/status-mobile/pull/13719)

Fixes #13246 

### Summary

Add missing actions (edit/delete/reply) to certain types of messages (emoji/sticker/audio)
To implement the needed functionalities there are 2 pull requests to this repo `status-mobile` and [status-go](https://github.com/status-im/status-go/pull/2771).

1. Editing emoji messages: Firstly, I had to allow the emoji content type to be edited from status-go. Secondly, when editing an emoji message, I had to check whether it is still an emoji message or is now a text message, and send the contentType. Also, in status-go I added a "ContentType" field to the "EditMessage" struct and updated the content type when a message is edited. Regarding UI, I had to make sure a white space is added at the end of an emoji message when edited (similar to text messages) to avoid overlapping with the "edited" text and icon.

2. Deleting emoji messages: added delete option on-long-press.

3. Audio message reply: In status-go, I added an AudioLocalURL field to the QuotedMessage struct. When an audio message is replied to, the AudioLocalURL field will be assigned to the QuotedMessage. In status-mobile, inside the quoted-message method, I check if the quoted message contains audio and if so, will render the audio message.

4. Sticker message reply: Initially, I attempted to do it in a similar way to replying to audio, but it didn't work as the sticker url is created using the sticker hash and not the message ID. Currently, the "responseTo" field includes the messageID only. So, what I did is simply get the sticker url from the message itself using the messageID inside the quoted-message method in status-mobile. I added a `HasSticker` field to the QuotedMessage struct in status-go to know if it is a sticker. Additionally, if the sticker is deleted, a message will be shown to the user stating "This message has been deleted!", which I think can also be implemented for replying to other content types.

5. Sticker message delete: added delete option on-long-press.

Sticker reply and delete also work in public chats.

### Review Notes
This PR depends on [that status-go PR](https://github.com/status-im/status-go/pull/2771) for backend functionality: 87db7d953e8fc748baa1fc56a7adffd8fada3055

#### Platforms

- Android
- iOS

status: ready

### CODE REVIEW

**Comment by OmarBasem** [comment]
To fix all the unnecessary formatting changes caused by the IDE, I had to open a new PR. Please follow on this PR. @flexsurfer @rasom @J-Son89 @briansztamfater @Parveshdhull

File: src/status_im/ui/screens/chat/message/message_old.cljs [hunk 1]
```diff
@@ -87,8 +87,9 @@
         timestamp-str]])))
 
 (defview quoted-message
-  [_ {:keys [from parsed-text image]} outgoing current-public-key public? pinned]
-  (letsubs [contact-name [:contacts/contact-name-by-identity from]]
+  [_ {:keys [from parsed-text image audio sticker id] :as message} outgoing current-public-key public? pinned chat-id]
+  (letsubs [contact-name [:contacts/contact-name-by-identity from]
+            replied-message (@(re-frame/subscribe [:chats/chat-messages chat-id]) id)] ;; message replied to
```
Comment (rasom) [inline src/status_im/ui/screens/chat/message/message_old.cljs]: As was suggested in the previous PR it makes sense to replace`defview` with defn and `letsubs` with `let`. `defview` is straightforward, adding a sample for `letsubs`:
```clojure
(let [contact-name @(re-frame/subscribe [:contacts/contact-name-by-identity from]])
      replied-message (get @(re-frame/subscribe [:chats/chat-messages chat-id]) id)]
...
```

Comment (OmarBasem) [reply src/status_im/ui/screens/chat/message/message_old.cljs]: Aha got it, thanks!

**Comment by flexsurfer** [comment]
i believe there was a status-go change in the first PR ?

File: src/status_im/ui/screens/chat/message/message_old.cljs [hunk 2]
```diff
@@ -617,11 +653,17 @@
...
-                                     :id       :delete}]
-                                   [])))
+           (fn []
+             (on-long-press
+              (concat
+               [{:on-press #(re-frame/dispatch [:chat.ui/reply-to-message message])
+                 :id       :reply
+                 :label    (i18n/label :t/message-reply)}]
+               (when (and outgoing config/delete-message-enabled?))
+               [{:on-press #(re-frame/dispatch [:chat.ui/soft-delete-message message])
```
Comment (flexsurfer) [inline src/status_im/ui/screens/chat/message/message_old.cljs]: `[{:on-press..` should be inside when form

**Comment by OmarBasem** [comment]
> i believe there was a status-go change in the first PR ?

@flexsurfer Yes, as mentioned above, this PR depends on [that status-go PR](https://github.com/status-im/status-go/pull/2771) for backend functionality. Commit hash: 87db7d953e8fc748baa1fc56a7adffd8fada3055

**Comment by OmarBasem** [comment]
I don't understand why it cannot be built now (for iOS) after rebasing.

`make lint` is passing
`make test` is passing
There are no conflicts.
App is building and running locally with no problem.

**Comment by flexsurfer** [comment]
hey @OmarBasem sorry for late response, could you please rebase, and try `make test` locally , seems like tests failed

**Comment by OmarBasem** [comment]
> hey @OmarBasem sorry for late response, could you please rebase, and try `make test` locally, seems like tests failed

@flexsurfer As mentioned above, I have already tried rebasing 3 times, but iOS build kept failing.
`make lint` was passing
`make test` was passing
There were no conflicts.
App was building and running locally with no problem.

**Comment by Parveshdhull** [comment]
>  but iOS build kept failing.

Hi @OmarBasem, there was problem with ios build, now fixed in https://github.com/status-im/status-mobile/pull/13912

Also, looks like other failed builds are due to the issue mentioned in https://github.com/status-im/status-mobile/pull/13620#discussion_r962210314 .
Because build succeed, its that below section is showing result of first build not current one.

This one is also fixed in https://github.com/status-im/status-mobile/pull/13926

Please can you try rebase now, builds should work fine.

**Comment by flexsurfer** [comment]
hey @OmarBasem you need to update status go version

**Comment by OmarBasem** [comment]
> hey @OmarBasem you need to update status go version

I see all checks have passed. It is the same version v0.107.1.

**Comment by flexsurfer** [comment]
but I suppose this PR depends on  https://github.com/status-im/status-go/pull/2771 ?  so version from PR should be used?

**Comment by OmarBasem** [comment]
> but I suppose this PR depends on [status-im/status-go#2771](https://github.com/status-im/status-go/pull/2771) ? so version from PR should be used?

@flexsurfer That's what I had when I first made the PR, but it was causing merge conflicts. Shall I update it now?

**Comment by flexsurfer** [comment]
yes, please

**Comment by OmarBasem** [comment]
> yes, please

@flexsurfer done

**Comment by pavloburykh** [comment]
hey @OmarBasem I've moved PR to the Review column as I see that it has not been reviewed yet. Please move it to E2E column as soon as it passes the review. After that, we will take PR into testing. Thank you!

**Comment by flexsurfer** [comment]
i know status-go has been merged, should we update and test this PR , @OmarBasem ?

**Comment by OmarBasem** [comment]
@flexsurfer yes we can do that

**Comment by OmarBasem** [comment]
@flexsurfer I am not sure if I still need to have it pointing to a specific status-go commit?

**Comment by flexsurfer** [comment]
i guess your changes already in latest go version so you could just use latest version

**Comment by OmarBasem** [comment]
> i guess your changes already in latest go version so you could just use latest version

@flexsurfer is `status-mobile` develop branch using latest `status-go` develop branch?

**Comment by flexsurfer** [comment]
i think yes

**Comment by OmarBasem** [comment]
> i think yes

Apparently not. The develop branch is currently using an older commit. I updated it to the corresponding commit.

**Comment by flexsurfer** [comment]
hey @OmarBasem please fix `make lint` , and also one question, is this PR can be tested in the new UI ?

**Comment by OmarBasem** [comment]
> hey @OmarBasem please fix `make lint` , and also one question, is this PR can be tested in the new UI ?

@flexsurfer What do you mean can be tested in the new UI? This PR is implementing old UI functionalities

**Comment by flexsurfer** [comment]
ok , @OmarBasem lets then merge it, because we don't test old ui anymore

**Review by flexsurfer — APPROVED** [review]

**Comment by flexsurfer** [comment]
but you need to bump version in status-go, so we can use it

**Comment by OmarBasem** [comment]
> but you need to bump version in status-go, so we can use it

Yes I already did

**Pull request merged**